packages/typescript-estree/src/parser.ts

```diff
@@ -195,11 +195,10 @@ function parseAndGenerateServices<T extends TSESTreeOptions = TSESTreeOptions>(
     };
   }
 
-  /**
-   * Generate a full ts.Program or offer provided instances in order to be able to provide parser services, such as type-checking
-   */
   const hasFullTypeInformation =
-    parseSettings.programs != null || parseSettings.projects.size > 0;
+    parseSettings.programs != null ||
+    parseSettings.projects.size > 0 ||
+    !!parseSettings.projectService;
```

Comment:
@yepitschunked following up on `parserSettings.typeAware`: whether the file is being parsed with type information is still inferable from existing `parseSettings` properties. So rather than refactor to add a _new_ property, this just corrects the logic for `const hasFullTypeInformation`.